src/AudioAnalysisTools/Events/EventExtentions.cs

```diff
@@ -360,5 +342,27 @@ public static double GetAverageAmplitudeInLowerNeighbourhood(SpectralEvent ev, d
 
             return filteredEvents;
         }
+
+        /// <summary>
+        /// Combines all the tracks in all the events in the passed list into a single track.
+        /// Each frame in the composite event is assigned the spectral point having maximum amplitude.
+        /// The points in the returned array are in temporal order.
+        /// TODO TODO WARNING!  This method needs to be generalised from WhipEvent - see first line.
+        /// </summary>
+        /// <param name="events">List of spectral events.</param>
+        public static IEnumerable<ISpectralPoint> GetCompositeTrack(List<EventCommon> events)
+        {
+            var spectralEvents = events.Select(x => (WhipEvent)x);
```

Comment:
If you make the method generic this is easily fixed, right?

```
public static IEnumerable<ISpectralPoint> GetCompositeTrac<T>k(List<T> events)
  where T : ITracks
```

Reply:
Yes, but I was not confident enough to do it.